### ISSUE TITLE
Functional models with list outputs do not should accept dict losses

### ISSUE THREAD

**Issue by divyashreepathihalli** [keras-team/keras#18708]
This is a feature gap
metrics argument in model.compile() accepts dict input for multiple outputs like follows 
output_metrics = {'output_1': 'categorical_crossentropy', 'output_2': 'categorical_crossentropy'}
model.compile(..., matrics=output_metrics)

we should support the same for loss

repro here 
https://colab.sandbox.google.com/drive/1DL4z2qm7XTJjlbfQQWjdQMlRJaU207QJ#scrollTo=TdNk9E9Knkt_

**Comment by haifeng-jin** [comment]
Updated colab link: https://colab.sandbox.google.com/drive/1Vx-etBP-pmltQ3-kFGAOWPa-KJm29dvQ

Seems a user error. Needs to call `model.evaluate()` with `y=[y_test, y_test]` instead of `y=y_test` since there are two heads of the model.

However, we do need to update the error message to avoid user confusion.
